python/ray/ml/config.py

```diff
@@ -105,6 +115,50 @@ def as_placement_group_factory(self) -> "PlacementGroupFactory":
         bundles = trainer_bundle + worker_bundles
         return PlacementGroupFactory(bundles, strategy=self.placement_strategy)
 
+    @classmethod
+    def validate_config(
+        cls,
+        config: Union[ScalingConfig, "ScalingConfigDataClass"],
+        allowed_keys: Iterable[str],
+        scaling_config_arg_name: str,
+        caller_name: str,
+    ):
+        """
+        Validate config by raising an exception if any of allowed_keys
+        differs from the default value.
+
+        Args:
+            config: Config to check.
+            allowed_keys: ScalingConfigDataClass attribute keys that
+                can have a value different than the default one.
+            scaling_config_arg_name: Name of the ScalingConfig argument to be used
+                in the exception message.
+            exc_obj_name: Name of the object calling this method to be used
+                in the exception message.
```

Comment:
As mentioned in the previous PR, let's remove these arguments. The enclosing method can re-raise the exception so that the affected key names are in the stack trace, but we should avoid passing strings just for raising error messages

Reply:
I wonder if we can return a list of bad keys instead. I think a single exception would be better from the user perspective than a long stack trace

Reply:
I would be open to raise a custom exception here that stores the keys as an attribute.

`ray.air.exceptions.ConfigError` or so